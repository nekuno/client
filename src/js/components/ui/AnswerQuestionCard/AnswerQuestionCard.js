import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './AnswerQuestionCard.scss';
import CardUser from "../../cardUsers/CardUser";
import translate from "../../../i18n/Translate";

@translate('AnswerQuestionCard')
export default class AnswerQuestionCard extends Component {

    static propTypes = {
        onClickHandler : PropTypes.func,
        question       : PropTypes.object,
        otherUser      : PropTypes.object,
        strings       : PropTypes.object
    };

    static contextTypes = {
        router: PropTypes.object.isRequired
    };

    handleClick(questionId) {
        const {otherUser} = this.props;
        this.context.router.push(`/answer-question/${questionId}/${otherUser.slug}`);
    }

    render() {
        const {children, question, strings} = this.props;
        return (
            <div className={styles.answerQuestionCard} onClick={this.handleClick.bind(this, question.questionId)}>
                <h3>{question.text}</h3>
                <p>{strings.answerQuestion}</p>
            </div>
        );
    }
}

AnswerQuestionCard.defaultProps = {
    strings: {
        answerQuestion      : 'Answer to see your answer',
    }
};