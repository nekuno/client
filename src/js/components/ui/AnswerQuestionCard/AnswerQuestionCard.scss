@import '../../../../scss/_partials/colors';

.answer-question-card {
  background-color: $purple-one;
  color: $white;
  -webkit-border-radius: 8px;
  -moz-border-radius: 8px;
  border-radius: 8px;
  margin: 8px 16px;
  padding: 4px 16px 8px 16px;
  h3 {
    font-size: 16px;
    font-weight: 600;
  }
  p {
    font-size: 14px;
  }
}