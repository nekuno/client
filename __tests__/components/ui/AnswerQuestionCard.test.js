import React from 'react';
import AnswerQuestionCard from '../../../src/js/components/ui/AnswerQuestionCard/AnswerQuestionCard.js';
import styles from '../../../src/js/components/ui/AnswerQuestionCard/AnswerQuestionCard.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test AnswerQuestionCard component', () => {
    const mockFn = jest.fn();
    const answerQuestionCard = shallow(<AnswerQuestionCard onClickHandler={mockFn}>Lorem ipsum</AnswerQuestionCard>);

    it('should be defined', () => {
        expect(answerQuestionCard).toBeDefined();
    });
    it('should render correctly', () => {
        expect(answerQuestionCard).toMatchSnapshot();
    });
    it('AnswerQuestionCard click event', () => {
        answerQuestionCard.find(`.${styles.answerQuestionCard}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
});
