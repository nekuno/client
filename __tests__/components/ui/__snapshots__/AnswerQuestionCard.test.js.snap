// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AnswerQuestionCard component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AnswerQuestionCard
    onClickHandler={[MockFunction]}
  >
    Lorem ipsum
  </AnswerQuestionCard>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          ¿Prefieres el mar o la montaña?
        </h3>,
        <p>
          Responde para ver su respuesta
        </p>,
      ],
      "className": "answerQuestionCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "¿Prefieres el mar o la montaña?",
        },
        "ref": null,
        "rendered": "¿Prefieres el mar o la montaña?",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Responde para ver su respuesta",
        },
        "ref": null,
        "rendered": "Responde para ver su respuesta",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            ¿Prefieres el mar o la montaña?
          </h3>,
          <p>
            Responde para ver su respuesta
          </p>,
        ],
        "className": "answerQuestionCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "¿Prefieres el mar o la montaña?",
          },
          "ref": null,
          "rendered": "¿Prefieres el mar o la montaña?",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Responde para ver su respuesta",
          },
          "ref": null,
          "rendered": "Responde para ver su respuesta",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
